### PR TITLE
[SYCL-MLIR] Add main() to `single_task.cpp` lit test

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Added a `main` function to first select a device, then invoke `host_single_task`, which contains the kernel `kernel_single_task`.
Added a run line to ensure the whole compilation, from source code to executable, is successful.
Added a run line to execute the executable on the host. 

Signed-off-by: Tsang, Whitney <whitney.tsang@intel.com>

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
depends on https://github.com/intel/llvm/pull/6967.

**Comment by whitneywhtsang** [comment]
Depends on https://github.com/intel/llvm/pull/7056

**Review by arnamoy10 — COMMENTED** [review]
Other than ettore's comments, LGTM!

File: polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp [hunk 1]
```diff
@@ -1,26 +1,34 @@
 // clang-format off
-// RUN: clang++ -fsycl -fsycl-device-only -emit-llvm -fsycl-targets=spir64-unknown-unknown-syclmlir %s -o %t.bc
+
+// TODO: Investigate and remove all "Warning"s.
+// RUN: clang++ -fsycl -fsycl-targets=spir64-unknown-unknown-syclmlir %s -o %t.out 2>&1 | FileCheck %s --implicit-check-not="{{warning|error|Error}}:"
```
[the remark this answers is not in the document]
Comment (whitneywhtsang) [reply polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp:4]: Added one for the host. Can I add one for device in the next PR? Looks like I need to modify `lit.cfg.py` to have device run working.

[the remark this answers is not in the document]
Comment (etiotto) [reply polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp:4]: Yes is fine to do it in a future PR.

**Review by etiotto — APPROVED** [review]

**Pull request merged**